expressServer/index.js

```diff
@@ -45,8 +49,9 @@ app.get('/api/snakes', async (_, res) => {
     const users = await User.find({ username: /snake/i });
     res.status(200).json(users);
   } catch (error) {
-    console.error('Error fetching snake users:', error);
     res.status(500).json({ message: 'Internal server error' });
+  } finally {
+    closeDB();
   }
```

Comment:
In the server, it's better to not close the connection, I can explain you this in a call if you prefer